plugins/provisioners/shell/provisioner.rb

```diff
@@ -135,24 +137,22 @@ def provision_winssh(args)
           # Upload the script to the machine
           @machine.communicate.tap do |comm|
             env = config.env.map{|k,v| comm.generate_environment_export(k, v)}.join(';')
-            shell = comm.machine_config_ssh.shell
-            if File.extname(upload_path).empty?
-              remote_ext = File.extname(path)[1..-1]
-              if !remote_ext
-                remote_ext = @machine.config.winssh.shell == "cmd" ? "bat" : "ps1"
-              end
-              upload_path << ".#{remote_ext}"
+            remote_ext = File.extname(path)
+            if remote_ext.empty?
+              remote_ext = @machine.config.winssh.shell == "cmd" ? ".bat" : ".ps1"
             end
+
+            remote_path = add_extension(upload_path(), remote_ext)
```

Comment:
The parens on `upload_path()` can be removed. Not included on method calls when no parameters are provided (it was my bad on the earlier comment).